install.yaml

```diff
@@ -38,21 +38,11 @@ post_install_actions:
     #ddev-generated
     # This is the second half of the trick that puts varnish "in front of" the web
     # container, just by switching the names.
-    {{- $project_tld := "ddev.site" -}}
-    {{- if .DdevGlobalConfig.project_tld }}{{ $project_tld = .DdevGlobalConfig.project_tld }}{{ end }}
-    {{- if .DdevProjectConfig.project_tld }}{{ $project_tld = .DdevProjectConfig.project_tld }}{{ end }}
-    {{- $novarnish_hostnames := print "novarnish." .DdevProjectConfig.name "." $project_tld -}}
-    {{- $sep := print "." $project_tld ",novarnish." -}}
-    {{- if .DdevProjectConfig.additional_hostnames }}
-    {{- $novarnish_hostnames = print $novarnish_hostnames "," "novarnish." (.DdevProjectConfig.additional_hostnames | join $sep) "." $project_tld -}}
-    {{- end }}
-    {{- if .DdevProjectConfig.additional_fqdns }}
-    {{- $novarnish_hostnames = print $novarnish_hostnames "," "novarnish." ( .DdevProjectConfig.additional_fqdns | join ",novarnish." ) -}}
-    {{- end }}
     services:
       web:
         environment:
-        - VIRTUAL_HOST={{ $novarnish_hostnames }}
+        {{- $novarnish_hostnames := splitList "," (env "DDEV_HOSTNAME") }}
+        - VIRTUAL_HOST={{ range $i, $n := $novarnish_hostnames }}novarnish.{{ replace (env "DDEV_TLD") "\\${DDEV_TLD}" (replace (env "DDEV_PROJECT") "\\${DDEV_PROJECT}" $n) }}{{ if lt (add1 $i) (len $novarnish_hostnames) }},{{ end }}{{ end }}
```

Comment:
Please add a comment here to explain what this is doing and why. It looks great and should completely solve this problem. 

The README needs to explain that if additional_hostnames or additional_fqdns changes, the `ddev add-on get` needs to be re-run.

It will need a new release as well.

Reply:
The comment was already here above the `services:`

I'm not very clear on how it works, but I've added more explanations there.